llvm/lib/Target/AMDGPU/AMDGPURewriteAGPRCopyMFMA.cpp

```diff
@@ -302,6 +302,10 @@ bool AMDGPURewriteAGPRCopyMFMAImpl::attemptReassignmentsToAGPR(
     const TargetRegisterClass *EquivalentAGPRRegClass =
         TRI.getEquivalentAGPRClass(MRI.getRegClass(InterferingReg));
 
+    // Do not reassign scale operands
+    if (EquivalentAGPRRegClass == &AMDGPU::AGPR_32RegClass)
```

Comment:
This doesn't seem like the right condition. It just happens the scale operands are the only 32-bit input case. This should more directly check the operand constraint instead of checking a hardcodes class equality